openmls/src/treesync/node.rs

```diff
@@ -16,3 +16,9 @@ pub enum Node {
     LeafNode(OpenMlsLeafNode),
     ParentNode(ParentNode),
 }
+
+/// Container enum with reference to a node in a tree.
+pub(crate) enum NodeReference<'a> {
+    Leaf(&'a OpenMlsLeafNode),
+    Parent(&'a ParentNode),
+}
```

Comment:
It is a bit annoying that we have to have an extra struct to accommodate these references, but I don't see a way to make it work like `&Node`.

Reply:
I find it quite practical tbh. And yes, I also don't see a way to combine it with Node (unless we do something super complicated).